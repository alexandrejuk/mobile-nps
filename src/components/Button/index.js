import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './style.module.css'

const Button = ({
  action,
  children,
  color,
  type,
}) => (
  <button
    onClick={action}
    className={classNames(
      styles.button,
      styles[color],
    )}
    type={type}
  >
    {children}
  </button>
)

Button.propTypes = {
  action: PropTypes.func,
  children: PropTypes.node.isRequired,
  color: PropTypes.oneOf([
    'outline',
    'primary',
  ]).isRequired,
  type: PropTypes.string,
}

Button.defaultProps = {
  action: null,
  type: 'button',
}

export default Button
