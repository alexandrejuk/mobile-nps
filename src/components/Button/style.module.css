@import '../../assets/css/colors.css';

.button {
  background-color: var(--color-california);
  border-radius: 3px;
  border: 1px solid var(--color-california);
  box-sizing: border-box;
  color: var(--color-white);
  cursor: pointer;
  font-family: 'Rubik', Arial, Helvetica, sans-serif;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
  transition: all 0.3s ease;
  padding: 8px 24px;
  width: 100%;
}

.button:active,
.button:focus {
  outline: none;
}

.button:active {
  opacity: 0.8;
}

.outline {
  background-color: var(--color-white);
  border: 1px solid var(--color-california);
  color: var(--color-california);
}

.button.outline:hover {
  background-color: var(--color-california);
  color: var(--color-white);
}

.button.outline:active {
  opacity: 0.8;
}
