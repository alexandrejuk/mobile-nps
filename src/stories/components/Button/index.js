import React from 'react'
import { action } from '@storybook/addon-actions'
import { Button } from '../../../components'

export default {
  title: 'Components/Button',
}

export const Primary = () => (
  <Button
    action={action('clicked')}
    color="primary"
  >
    Primary Button
  </Button>
)

export const Outline = () => (
  <Button
    action={action('clicked')}
    color="outline"
  >
    Outline Button
  </Button>
)
