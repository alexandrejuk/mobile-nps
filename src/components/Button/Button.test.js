import React from 'react'
import { render } from '@testing-library/react'

import Button from './'

describe('Button component', () => {
  it('should render correctly primary button', () => {
    const { container } = render(
      <Button
        color="primary"
      >
        Button component
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly outline button', () => {
    const { container } = render(
      <Button
        color="outline"
      >
        Button component
      </Button>
    )
    expect(container).toMatchSnapshot()
  })
})
