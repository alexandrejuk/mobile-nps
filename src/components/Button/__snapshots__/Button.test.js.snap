// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly outline button 1`] = `
<div>
  <button
    class="button outline"
    type="button"
  >
    Button component
  </button>
</div>
`;

exports[`Button component should render correctly primary button 1`] = `
<div>
  <button
    class="button primary"
    type="button"
  >
    Button component
  </button>
</div>
`;
